samples/ios/Continote/Continote/MyNotesViewController.swift

```diff
@@ -15,39 +15,155 @@
-    // Style all labels.
-    temporaryTodoLabel.applyAppTheme(for: .normalText)
+    switch segueIdentifier {
+    case .editNote:
+      // We need to let the EditNoteViewController know which Note the user wishes to edit
+      // before the segue can be performed.
+      let cell = sender as! MyNotesTableViewCell
+      let editNoteViewController = segue.destination as! EditNoteViewController
+      editNoteViewController.keyOfNoteToEdit = cell.noteKey
+    }
+  }
+
+  override func handleUserSignedIn(_ user: User) {
+    super.handleUserSignedIn(user)
+
+    // Set up our TableView up to sync with the Notes for the current user from the Firebase
+    // Realtime Database.
+    notesFirebaseDatabaseRef = Database.database().reference().child("notes").child(user.uid)
+    myNotesTableViewDataSource = MyNotesTableViewDataSource(
+      query: notesFirebaseDatabaseRef!,
+      populateCell: { tableView, indexPath, snapshot in
+        // Get the data for this Note, as gathered from the Firebase Realtime Database, and parse
+        // it to create a Note struct for this cell.
+        let noteKey:String = snapshot.key
+        let note: Note? = Note(with: snapshot)
+
+        // Get and then populate a cell in the TableView to use for this Note.
+        let cell = tableView.dequeueReusableCell(withIdentifier: "MyNotesTableViewCell")
+                    as! MyNotesTableViewCell
+        cell.noteKey = noteKey
+        cell.note = note
+
+        return cell
+      })
+
+    myNotesTableViewDataSource?.bind(to: myNotesTableView)
+
+    // Finally, show this screen's UI since everything is ready.
+    myNotesTableView.isHidden = false
+    writeNewNoteButton.isHidden = false
   }
 
   override func handleUserSignedOut() {
     super.handleUserSignedOut()
 
+    // Hide this screen's UI since the user must be signed in to see any Notes.
+    myNotesTableView.isHidden = true
+    writeNewNoteButton.isHidden = true
+
     // The user must be signed in to view this screen, so navigate away from it if the user is
     // signed out.
     navigationController?.popViewController(animated: true)
   }
+
+  /**
+   Handles when the user taps the "write a new note" button.
+
+   Adds a new Note to the Firebase Realtime Database for the current user,
+   then opens that Note to allow the user to begin writing.
+
+   - Parameter sender: The object that called this action. This should only be the
+   writeNewNoteButton.
+   */
+  @IBAction func writeNewNoteButtonAction(_ sender: Any) {
+    guard let notesFirebaseDatabaseRef = notesFirebaseDatabaseRef else {
+      // This should never happen because the user must be signed in, but just in case.
+      return
+    }
+
+    // Add a new, empty Note to the Firebase Realtime Database for the current user.
+    let newNote: Note = Note(title: "", content: "")
+    let newNoteRef: DatabaseReference = notesFirebaseDatabaseRef.childByAutoId()
+    newNoteRef.setValue(newNote.asFirebaseData) { (error, ref) -> Void in
```

Comment:
You can pass in `self` here with: `) [weak self] (error, ref) -> Void in`. This should be used since you're accessing it below with `performSegue`, which will turn that into a `self?.` call.